src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -98,11 +117,13 @@
     private static final int MAX_POTIONTYPE_ID = Integer.MAX_VALUE >> 5; // Int (SPacketEffect)
     private static final int MAX_ENCHANTMENT_ID = Short.MAX_VALUE - 1; // Short - serialized as a short in ItemStack NBTs.
     private static final int MAX_ENTITY_ID = Integer.MAX_VALUE >> 5; // Varint (SPacketSpawnMob)
+    private static final int MAX_FLUID_ID = 4096; //There are only 4096 possible blocks
```

Comment:
or 4095 like blocks?

Reply:
No, this should really be its own constant like it is now. Changing the maximum block id in the future should not magically change the maximum fluid id.

Reply:
What other reason is there to limit the number of fluids though? If it should be its own constant, then it shouldn't be defined as 4096.